### DISCUSSION TITLE
ADT A28 and A31 Message Type missing in V251

### DISCUSSION THREAD

**Discussion by safwan4624** [nHapiNET/nHapi#354] [General]
Hi,
I'm a new user of nhapi.
Could someone please explain how to create and parse ADT A28 and A31 messages in V251?

Thanks in advance
Safwan

**Comment by milkshakeuk** [comment]
hi @safwan4624 have you checked the wiki [Getting Started](https://github.com/nHapiNET/nHapi/wiki/Getting-Started), [Encoding a message](https://github.com/nHapiNET/nHapi/wiki/Encoding-a-message), [Decoding a message](https://github.com/nHapiNET/nHapi/wiki/Decoding-a-message)?
There are also some [older tutorials](https://saravanansubramanian.com/hl7tutorials/#hl7-programming-tutorials-using-nhapi-and-net) on the web which are still useful, perhaps have a look there as well.

If you are still having after reviewing the above issues please come back and continue this thread.

**Comment by milkshakeuk** [comment]
@safwan4624 I just realised this might also be pertinent to your question.
https://github.com/nHapiNET/nHapi/wiki/Decoding-a-message#event-map-to-code-structure-mappings
